### PR TITLE
Bump flink 1.13.1

### PR DESCRIPTION
**Pull request by jianyun8023**
Bump flink 1.13.1

Change:
- `PulsarCatalogFactory` use new API.
- `TableSchema` is partially updated to new API `ResolvedSchema`

### CODE REVIEW

**Comment by nlu90** [comment]
Any concrete user request for bumping to Flink 1.13?

Currently our cloud Flink is on 1.12. And 1.13 is just released, which I think we should wait for some time.

File: pulsar-flink-connector/src/main/java/org/apache/flink/streaming/util/serialization/PulsarDeserializationSchemaBuilder.java [hunk 1]
```diff
@@ -60,9 +58,6 @@ public PulsarDeserializationSchemaBuilder() {
     }
 
     public PulsarDeserializationSchema<V> build() {
-        if (dataType == null) {
-            dataType = DataTypes.RAW(TypeInformationUtils.getTypesAsRow(recordClass)).bridgedTo(recordClass);
-        }
-        return new PulsarDeserializationSchemaWrapper<>(valueDeserializer, dataType);
+        throw new UnsupportedOperationException("PulsarDeserializationSchemaBuilder is deprecated, use PulsarDeserializationSchema#valueOnly.");
```
Comment (syhily) [inline pulsar-flink-connector/src/main/java/org/apache/flink/streaming/util/serialization/PulsarDeserializationSchemaBuilder.java:61]: Should we return the new `PulsarDeserializationSchema` instead.

**Comment by jianyun8023** [comment]
@nlu90 Some domestic users need to use the new Flink 1.13, so an upgrade is needed. Now the new features will continue to be developed in 1.12, don't worry.

**Review by syhily — APPROVED** [review]

File: pulsar-flink-connector/src/main/java/org/apache/flink/streaming/util/serialization/PulsarDeserializationSchemaBuilder.java (hunk 1 above)
Comment (jianyun8023) [reply pulsar-flink-connector/src/main/java/org/apache/flink/streaming/util/serialization/PulsarDeserializationSchemaBuilder.java:61]: This class was marked as `deprecated` in 1.12. In upgrade 1.13, re-implementing it also needs to re-study how to implement it. Or we remove it in 1.13.

**Pull request merged**